addons/binding/org.openhab.binding.camera/src/main/java/org/openhab/binding/camera/handler/CameraHandler.java

```diff
@@ -0,0 +1,195 @@
+                refreshData();
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        if (logger.isDebugEnabled()) {
+            logger.debug("Initialize thing: {}::{}", getThing().getLabel(), getThing().getUID());
+        }
+        Object paramUrl = getConfig().get("urlSnapshot");
+        urlSnapshot = String.valueOf(paramUrl);
+        Object paramUsername = getConfig().get("urlUsername");
+        if (paramUsername != null) {
+            urlUsername = String.valueOf(paramUsername);
+        }
+        Object paramPassword = getConfig().get("urlPassword");
+        if (paramPassword != null) {
+            urlPassword = String.valueOf(paramPassword);
+        }
+        try {
+            Object param = getConfig().get("poll");
+            CONFIGURATION_POLLTIME_MS = (int) (Double.parseDouble(String.valueOf(param)) * 1000);
+            logger.debug("Schedule update at fixed rate {} ms.", CONFIGURATION_POLLTIME_MS);
+        } catch (Exception e1) {
+            logger.warn("could not read poll time", e1);
+        }
+        if (initialized.compareAndSet(false, true)) {
+            WeakReference<CameraHandler> weakReference = new WeakReference<>(this);
+            serviceCached.submit(new Callable<Void>() {
+                @Override
+                public Void call() throws Exception {
+                    while (weakReference.get() != null) {
+                        try {
+                            refreshData();
+                        } catch (Exception e) {
+                            logger.error("error in refresh", e);
+                        }
+                        Thread.sleep(Math.max(10, CONFIGURATION_POLLTIME_MS));
+                    }
+                    return null;
+                }
+            });
+        }
+        updateStatus(ThingStatus.OFFLINE);
+        // Note: When initialization can NOT be done set the status with more details for further
+        // analysis. See also class ThingStatusDetail for all available status details.
+        // Add a description to give user information to understand why thing does not work
+        // as expected. E.g.
+        // updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+        // "Can not access device as username and/or password are invalid");
+    }
+
+    private void refreshData() {
+        if (refreshInProgress.compareAndSet(false, true)) {
+            try {
+                for (Channel cx : getThing().getChannels()) {
+                    if (cx.getAcceptedItemType().equals("Image")) {
+                        if (logger.isTraceEnabled()) {
+                            logger.trace("Will update: {}::{}::{}", getThing().getUID().getId(),
+                                    cx.getChannelTypeUID().getId(), getThing().getLabel());
+                        }
+                        if (urlSnapshot != null) {
+                            try {
```

Comment:
Rather split this complete method of and consider the using of the `HttpUtil` as mentioned earlier.

Reply:
See comment below.